org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/LanguageServerImpl.java

```diff
@@ -263,6 +266,7 @@ protected ServerCapabilities createServerCapabilities(InitializeParams params) {
 		serverCapabilities.setDocumentFormattingProvider(true);
 		serverCapabilities.setDocumentRangeFormattingProvider(true);
 		serverCapabilities.setDocumentHighlightProvider(true);
+		serverCapabilities.setFoldingRangeProvider(true);
```

Comment:
The server capabilities should not be set if we expect that we may see an UnsupportedOperationException. Maybe check if any of the available languages has a folding provider?